In the case of a non-matched route, shouldCache should be false

jonathanmaron opened the pull request.

bramstroker commented on src/StrokerCache/Strategy/CacheAllExcept.php:
```diff
@@ -27,48 +28,54 @@ public function shouldCache(MvcEvent $event)
 
-        $controller  = $this->normalize($routeMatch->getParam('controller'));
-        $action      = $this->normalize($routeMatch->getParam('action'));
+        if ($routeMatch instanceof RouteMatchHttp || $routeMatch instanceof RouteMatchConsole) {
```
I think we can also check for $routeMatch !== null here

bramstroker commented:
Thanks for PR, Could you also add a unit test please?

bramstroker commented:
I'll have a look into the failing travis builds tomorrow.

jonathanmaron commented:
I have implemented both your requests.

I have been taught from a very early age that a function should only return in one place, hence I personally disagree with your suggest to introduce a second return in the function. However, I respect it is your code base, and thus you can make make that call. I have implemented: 

    if (null === $routeMatch) {
        return false;
    }

I have also implemented the unit test testShouldCacheWhenRouteMatchIsNull() which proves the changes work as expected.

jonathanmaron commented:
The Strategy

    StrokerCache/Strategy/Controller.php

suffered from the same problem as 

    StrokerCache/Strategy/CacheAllExcept.php

hence I fixed that too.

Interestingly, 

    StrokerCache/Strategy/Route.php

does not suffer from the same problem.

bramstroker commented:
About the return early; this is very much personal taste. Both approaches have there pro's and cons and there is really no wrong or right. I have seen both applied a lot in the industry. Personally I prefer to return early because you can first get rid of your invalid cases before entering the "main" function body. As a bonus you don't execute unnecessary code and you have less nested if's which makes the code more readable.

Thanks for your efforts in this PR.

jonathanmaron commented:
Returning early is certainly a matter of personal taste, and I also have seen code bases using both approaches.

Once you have merged the pull request, would you consider releasing a new version? I would like to remove the dev-master dependency in my composer.json file.

bramstroker commented:
Yes sure. I'll tag a new stable release when I've merged this PR. I'm going to fix the travis build within the next 2 hours. Than I'll merge if all unit tests pass.

jonathanmaron commented:
Thank you. I just installed 0.3.1 into my project. All good. :-)
